cmake/Sanitizers.cmake

```diff
@@ -35,6 +35,24 @@ function(enable_sanitizers project_name)
     list(JOIN SANITIZERS "," LIST_OF_SANITIZERS)
 
   endif()
+  
+  
+if (ENABLE_SANITIZER_ADDRESS AND (ENABLE_SANITIZER_THREAD OR ENABLE_SANITIZER_MEMORY))
+    message(FATAL_ERROR "SANITIZER_ADDRESS is not compatible with "
+        "SANITIZER_THREAD or SANITIZER_MEMORY.")
+endif ()
+
+
+if (ENABLE_SANITIZER_THREAD AND (ENABLE_SANITIZER_ADDRESS OR ENABLE_SANITIZER_MEMORY))
+    message(FATAL_ERROR "ENABLE_SANITIZER_THREAD is not compatible with "
+        "ENABLE_SANITIZER_ADDRESS or SANITIZER_MEMORY.")
+endif ()
+
+
+if (ENABLE_SANITIZER_MEMORY AND (ENABLE_SANITIZER_THREAD OR ENABLE_SANITIZER_ADDRESS))
+    message(FATAL_ERROR "ENABLE_SANITIZER_MEMORY is not compatible with "
+        "ENABLE_SANITIZER_ADDRESS or SANITIZER_MEMORY.")
```

Comment:
@davex98 I think this should read "ENABLE_SANITIZER_ADDRESS or SANITIZER_THREAD"?

Reply:
Couldn't we do something equivalent but less redundent?
if (ENABLE_SANITIZER_ADDRESS)
    if (ENABLE_SANITIZER_THREAD OR ENABLE_SANITIZER_MEMORY)
        message(FATAL_ERROR "SANITIZER_ADDRESS is not compatible with "
        "SANITIZER_THREAD or SANITIZER_MEMORY.")
    endif ()
elseif (ENABLE_SANITIZER_THREAD AND ENABLE_SANITIZER_ADDRESS)
    message(FATAL_ERROR "ENABLE_SANITIZER_THREAD is not compatible with SANITIZER_MEMORY.")
endif ()